prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -346,6 +346,22 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
         return itemLineages
     }
 
+    /**
+     * Fetch an action ID for a given [reportId].
+     * @param txn an optional database transaction
+     * @return an action ID, or null if no action ID was found
+     */
+    fun fetchActionIdForReport(
+        reportId: UUID,
+        txn: DataAccessTransaction? = null
+    ): Long? {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        return ctx.selectDistinct(REPORT_LINEAGE.ACTION_ID)
```

Comment:
Why use the report lineage table rather than the report_file table?

In the future it's technically possible for multiple report lineage entries to have the same parent id with different actions, which would make this unstable.

Reply:
Oh, I did not see it had the report ID.  Sooooooo, what is the difference?

Reply:
I think the big one is that this requires the `distinct` which will result in a potentially unreliable state. Where as the `report_id` in the `report_file` table should both be indexed already (I think) and unique

Reply:
Addition: report_file has the report ID as PK.  report_lineage could end up having more than one entry for a report ID.  Furthermore, why do these tables not have FKs????

Reply:
@whytheplatypus Done.  Thx